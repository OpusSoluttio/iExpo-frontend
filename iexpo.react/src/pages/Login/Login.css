*{
    margin : 0;
    padding : 0;
    font-family : "Baloo 2"
}

@font-face{
    font-family: "Baloo 2";
    src: url("../../assets//fonts//Baloo2/Baloo2-Regular.ttf");

    font-family: "Baloo 2 Medium";
    src: url("../../assets//fonts//Baloo2/Baloo2-Medium.ttf");

    font-family: "Baloo 2 SemiBold";
    src: url("../../assets//fonts//Baloo2/Baloo2-SemiBold.ttf");

    font-family: "Baloo 2 Bold";
    src: url("../../assets//fonts//Baloo2/Baloo2-Bold.ttf");

    font-family: "Baloo 2 ExtraBold";
    src: url("../../assets//fonts//Baloo2/Baloo2-ExtraBold.ttf");
}



.Login{
    height: 100vh;
    width: 100%;
    background-image: linear-gradient(to bottom right, #97d3f3, #015281);
    display: flex;
    justify-content: center;
    align-items: center;
}

.Login main{
    display: flex;
    flex-flow: column nowrap;
    justify-content: space-between;
    align-items: center;
    width: 30%;
    max-height: 80vh;
    margin-top: -3em;
}


.logo{
    width: 55%;
}

.Login main p{
    text-align: center;
    color: #FFF;
    font-size: 2.5em;
    height: 100%;
    margin: -0.3em 0 0.5em 0;
}

.login-form{
    width: 93%;
    height: 14em;
    display: flex;
    flex-flow: column nowrap;
    justify-content: space-between;
}

.text-input{
    outline: none;
    padding: 0;
    border: 0;
    border-bottom: 3px solid #00324D;
    background: none;
    color: #fff;
    font-family: "Baloo 2",sans-serif;
    font-size: 1.7em;
}

                                                        /* FIX ME */

.efeito ~ .focus-border{
    position: relative;
    bottom: 7%;
    left: 50%;
    width: 0;
    height: 4px;
    background-color: #FFF;
    border-bottom:0px;
    transition: 0.4s;
}
.efeito:focus ~ .focus-border{
    outline: none;
    width: 100%;
    transition: 0.6s; 
    left: 0;
}

.efeito_2 ~ .focus-border_2{
    position: relative;
    bottom: 7%;
    left: 50%;
    width: 0;
    height: 4px;
    background-color: #fff;
    transition: 0.4s;
}
.efeito_2:focus ~ .focus-border_2{
    outline: none;
    width: 100%;
    transition: 0.6s; 
    left: 0;
}




.text-input::placeholder{
    color: #FFFFFF;
}

.submit{
    padding: 0.25em;
    background-color : #00324D;
    color: #fff;
    font-size: 2em;

    border: 0.1px solid #00324D;
    border-radius: 5px;
    transition: ease-in-out 0.2s;

}

.submit:hover{
    filter: brightness(160%);
    box-shadow: #ffffff73 3px 5px 10px;
    /* color: #00324D; */
}

.error-modal{
    min-height: 10vh;
    min-width: 100vh;
}

.error-title{
    color: #99111a;
    font-family: "Baloo 2 ExtraBold";
    font-size: 2em;
}

.error-description{
    text-align: justify;
}

@media only screen and (max-width: 1700px){
    .Login main{
        width: 30%;
    }

    .Login main p{
        font-size: 2.3em;
    }
}


@media only screen and (max-width: 1450px){
    .Login main p{
        font-size: 1.6em;
    }

    .Login main{
        width: 25%;
    }

    /* .logo{
        width: 50%;
    } */

    .login-form{
        width: 90%;
        height: 9em;
    }

    .text-input{
        font-size: 1.2em;
        border-bottom: 3px solid #00324D;
    }

    .submit{
        font-size: 1.2em;
    }
}

@media only screen and (max-width: 1290px){
    .Login main p{
        font-size: 1.3em;
    }
}


@media only screen and (max-width: 1100px){
    .Login main{
        width: 50%;
    }

    .Login main p{
        font-size: 2.5em;
    }

    .logo{
        width: 60%;
    }

    .login-form{
        width: 100%;
        height: 15em;
    }

    .submit{
        font-size: 1.75em;
    }

    .text-input{
        font-size: 2em;
    }

    .error-modal{
        min-height: 20vh;
        min-width: 40vh;
    }
}


@media only screen and (max-width: 800px){
    .Login main p{
        font-size: 1.95em;
    }

    .login-form{
        height: 12em;
    }

    .text-input{
        font-size: 1.5em;
    }
}


@media only screen and (max-width: 500px){
    .Login main{
        width: 80%;
    }

    .Login main p{
        font-size: 1.7em;
    }

    .login-form{
        height: 9em;
    }

    .text-input{
        font-size: 1.2em;
        border-bottom: 3px solid #00324D;
    }

    .submit{
        font-size: 1.2em;

    }
}

@media only screen and (max-width: 400px){
    .Login main p{
        font-size: 1.4em;
    }
}





 /* PARA O CELULAR DEITADO */
@media only screen and (max-height: 500px){
    .Login main{
        flex-flow: row nowrap;
        width: 70%;
    }

    .Login main p{
        display: none;
    }

    .logo{
        width: 33%;
        margin: 0 2em;
    }

    .login-form{
        width: 40%;
    }
}



/* & label.Mui-focused': {
    color: 'green',
  },
  '& .MuiInput-underline:after': {
    borderBottomColor: 'green',
  },
  '& .MuiOutlinedInput-root': {
    '& fieldset': {
      borderColor: 'red',
    },
    '&:hover fieldset': {
      borderColor: 'yellow',
    },
    '&.Mui-focused fieldset': {
      borderColor: 'green', */