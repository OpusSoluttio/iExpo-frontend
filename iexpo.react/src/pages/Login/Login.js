import React, { Component } from "react";
import Logo from "../../assets/img/logo.png";
// import TextField from '@material-ui/core/TextField';
// import TextInput from "react-materialize";
import Modal from "react-responsive-modal";
import ErrorModal from "../../components/ErrorModal"
import "./Login.css";
import Axios from 'axios';


export default class Login extends Component {
    constructor() {
        super();
        this.state = {
            email: "",
            senha: "",
            erro: false,
            mensagemRetorno: "",


            // FIX ME
            // disableSubmit : false
        }
    }

    /// função chamada quando aperta o "X" do modal
    fecharModal = () => {
        this.setState({ erro: false });
    }

    atualizarInputs = () => {
        this.setState({ email: document.querySelector("#email").value });
        this.setState({ senha: document.querySelector("#senha").value });
    }

    efetuarLogin = (event) => {
        event.preventDefault();
        // this.setState({disabled : true})

        if (this.state.email === null || this.state.email === "") {
            this.setState({ erro: true });
            this.setState({ mensagemRetorno: "Preencha o campo de email corretamente" });
        } else if (this.state.senha === null || this.state.senha === "") {
            this.setState({ erro: true });
            this.setState({ mensagemRetorno: "Preencha o campo da senha corretamente" });
        } else {


            Axios.post("http://26.98.132.83:5000/api/login", {
                email: this.state.email,
                senha: this.state.senha
            })
                .then(response => {
                    if (response.status === 200) {
                        localStorage.setItem("usuario-iexpo", response.data.token);
                        this.props.history.push('/')
                    }else if (response.status === 404){
                        this.setState({ erro: true, mensagemRetorno: "Usuário ou Senha inválidos" });
                    } else {
                        console.log("Algo deu errado.");
                        this.setState({ erro: true, mensagemRetorno: "Ocorreu um erro inesperado! Por favor, tente novamente mais tarde" });

                    }
                })
                .catch(erro => {
                    this.setState({ erro: true, mensagemRetorno: "Ocorreu um erro inesperado! Por favor, tente novamente mais tarde" });
                    console.log(erro);
                });
        }
    }
    //TODO
    //preencher com url certinha da api e com qual rota vai ser redirecionado.
    //depois de testado retirar os console.log


    render() {
        return (
            <div className="Login">
                <main>

                    <Modal
                        className="error-alert"
                        open={this.state.erro}
                        onClose={this.fecharModal}
                        focusTrapped={false}
                        center={true}
                        children={ <ErrorModal mensagemRetorno={this.state.mensagemRetorno} /> }
                    />

                    <img src={Logo} className="logo" alt="Logo da iExpo" />
                    <p>Sistema de administradores</p>

                    <form className="login-form" onSubmit={this.efetuarLogin} method="POST">
                        <input type="email"
                            className="text-input efeito"
                            autoComplete="off"
                            onChange={this.atualizarInputs}
                            id="email"
                            name="email"
                            placeholder="Email"
                            required />

                            <span className="focus-border" />

                        <input type="password"
                            className="text-input efeito_2"
                            onChange={this.atualizarInputs}
                            id="senha"
                            name="senha"
                            placeholder="Senha"
                            required />

                            <span className="focus-border_2" />
                        <input
                            type="submit" 
                            className="submit" 
                            value="Entrar"/>
                    </form>
                </main>
            </div>
        )
    }
}

// const theme = createMuiTheme({
//     overrides: {
//       // Nome da folha de estilo
//       MuiTextField, ⚛️
//       MuiButton: {
//         // Nome da regra
//         text: {
//           // Algum CSS
//           color: 'white',
//         },
//       },
//     },
//   });