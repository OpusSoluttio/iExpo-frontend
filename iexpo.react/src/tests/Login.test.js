import React from "react";
import Enzyme, {shallow, mount, render} from "enzyme";
import Login from "../pages/Login/Login";
import renderer from "react-test-renderer";
import Adapter from 'enzyme-adapter-react-16';

Enzyme.configure({ adapter: new Adapter() })

describe("Testa página de Login",() => {

    it("Compara componente ao snapshot",() => {
        const componente = renderer.create( <Login />).toJSON();
        expect(componente).toMatchSnapshot();
    });

    it("Testa estado inicial do componente",() => {
        const componente = mount( <Login />);
        expect(componente.state("carregando")).toEqual(false);
        expect(componente.state("erro")).toEqual(false);
        expect(componente.state("email")).toEqual("");
        expect(componente.state("senha")).toEqual("");
    });

    it("Testa mudança de estado dos inputs de email e senha",() =>{
        const componente = mount( <Login /> );
        
        // componente.find("#email").simulate("change",{
        //     target: {value : "email@email.com"}
        // });

        componente.find('input').at(0).simulate('change', { target: { name: 'email', value: "email@email.com" } });
        componente.find('input').at(1).simulate('change', { target: { name: 'senha', value: "12345678" } });

        expect(componente.state("email")).toEqual("email@email.com");
        expect(componente.state("senha")).toEqual("12345678");
    });

    it("Testa mudança de estado  para carregando ao clicar no submit",() =>{
        const componente = mount( <Login />);
        componente.setState({email : "email@email.com"});
        componente.setState({senha : "12345678"});

        componente.find("form").simulate("submit");

        expect(componente.state("carregando")).toEqual(true);
    })

})