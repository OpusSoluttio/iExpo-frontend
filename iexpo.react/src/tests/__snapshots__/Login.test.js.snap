// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testa p√°gina de Login Compara componente ao snapshot 1`] = `
<div
  className="Login"
>
  <main>
    <img
      alt="Logo da iExpo"
      className="logo"
      src="logo.png"
    />
    <p>
      Sistema de administradores
    </p>
    <form
      className="login-form"
      method="POST"
      onSubmit={[Function]}
    >
      <input
        autoComplete="off"
        className="text-input efeito"
        id="email"
        maxLength={100}
        name="email"
        onChange={[Function]}
        placeholder="Email"
        required={true}
        type="email"
      />
      <span
        className="focus-border"
      />
      <input
        className="text-input efeito_2"
        id="senha"
        maxLength={60}
        minLength={6}
        name="senha"
        onChange={[Function]}
        placeholder="Senha"
        required={true}
        type="password"
      />
      <span
        className="focus-border_2"
      />
      <input
        className="submit"
        type="submit"
        value="Entrar"
      />
    </form>
  </main>
</div>
`;
